### PR TITLE
pythonPackages.webargs: init at 6.1.0

### PR DESCRIPTION
**Pull request by cript0nauta**
###### Motivation for this change

I added the webargs and webtest-aiohttp Python packages. Also, removed some unused dependencies from marshmallow (otherwise the webargs test would fail).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ buildPythonPackage, fetchFromGitHub, lib, isPy27, webtest, invoke, flake8
+, aiohttp, pytest, pytest-aiohttp }:
```
Comment (jonringer) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:2]:
```suggestion
, aiohttp, pytest, pytest-aiohttp }:

```

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "sloria";
+    repo = pname;
+    rev = version;
+    sha256 = "1apr1x0wmnc6l8wv67z4dp00fiiygda6rwpxlspfk7nk9zz37q2j";
+  };
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:15]: default for buildPythonPackage
```suggestion
```

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  propagatedBuildInputs = [ webtest ];
+  checkInputs = [ invoke flake8 aiohttp pytest pytest-aiohttp ];
+
+  meta = {
+    description =
+      "Provides integration of WebTest with aiohttp.web applications";
+    homepage = "https://github.com/sloria/webtest-aiohttp";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ cript0nauta ];
```
Comment (jonringer) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:28]: personal taste
```suggestion
  meta = with lib; {
    description =
      "Provides integration of WebTest with aiohttp.web applications";
    homepage = "https://github.com/sloria/webtest-aiohttp";
    license = licenses.mit;
    maintainers = with maintainers; [ cript0nauta ];
```

File: pkgs/development/python-modules/webargs/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ buildPythonPackage, fetchPypi, lib, isPy27, marshmallow, pytest
+, pytest-aiohttp, webtest, webtest-aiohttp, flask, django, bottle, tornado
+, pyramid, falcon, aiohttp }:
```
Comment (jonringer) [inline pkgs/development/python-modules/webargs/default.nix:3]:
```suggestion
, pyramid, falcon, aiohttp }:

```

File: pkgs/development/python-modules/webargs/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  pname = "webargs";
+  version = "6.1.0";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0gxvd1k5czch2l3jpvgbb53wbzl2drld25rs45jcfkrwbjrpzd7b";
+  };
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/webargs/default.nix:14]: default for buildPythonPackage
```suggestion
```

File: pkgs/development/python-modules/webargs/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    aiohttp
+  ];
+
+  meta = {
+    description =
+      "Declarative parsing and validation of HTTP request objects, with built-in support for popular web frameworks";
+    homepage = "https://github.com/marshmallow-code/webargs";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ cript0nauta ];
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/webargs/default.nix:41]:
```suggestion
  meta = with lib; {
    description =
      "Declarative parsing and validation of HTTP request objects, with built-in support for popular web frameworks";
    homepage = "https://github.com/marshmallow-code/webargs";
    license = licenses.mit;
    maintainers = with maintainers; [ cript0nauta ];
  };
```

**Comment by cript0nauta** [comment]
@jonringer thanks for all the suggestions! I applied them all and rebased the branch.

**Comment by jonringer** [comment]
@cript0nauta still interested in this? sorry it got backburnered

File: pkgs/development/python-modules/webargs/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    bottle
+    tornado
+    pyramid
+    falcon
+    aiohttp
+  ];
+
+  meta = with lib; {
+    description =
+      "Declarative parsing and validation of HTTP request objects, with built-in support for popular web frameworks";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webargs/default.nix:37]:
```suggestion
    description = "Declarative parsing and validation of HTTP request objects, with built-in support for popular web frameworks";
```

File: pkgs/development/python-modules/webargs/default.nix [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  checkPhase = ''
+    rm tests/test_webapp2parser.py  # webapp2 doesn't support python 3
+    pytest
+  '';
+
+  propagatedBuildInputs = [ marshmallow ];
+  checkInputs = [
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webargs/default.nix:22]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/webargs/default.nix [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0gxvd1k5czch2l3jpvgbb53wbzl2drld25rs45jcfkrwbjrpzd7b";
+  };
+
+  checkPhase = ''
+    rm tests/test_webapp2parser.py  # webapp2 doesn't support python 3
+    pytest
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webargs/default.nix:18]:
```suggestion
  preCheck = ''
    rm tests/test_webapp2parser.py  # webapp2 doesn't support python 3
  '';
```

File: pkgs/development/python-modules/webargs/default.nix [hunk 10]
```diff
@@ -0,0 +1,42 @@
+{ buildPythonPackage, fetchPypi, lib, isPy27, marshmallow, pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webargs/default.nix:1]:
```suggestion
{ buildPythonPackage, fetchPypi, lib, isPy27, marshmallow, pytestCheckHook
```

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sloria";
+    repo = pname;
+    rev = version;
+    sha256 = "1apr1x0wmnc6l8wv67z4dp00fiiygda6rwpxlspfk7nk9zz37q2j";
+  };
+
+  checkPhase = ''
+    invoke test
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:17]: Can we use pytestCheckHook instead?

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  '';
+
+  propagatedBuildInputs = [ webtest ];
+  checkInputs = [ invoke flake8 aiohttp pytest pytest-aiohttp ];
+
+  meta = with lib; {
+    description =
+      "Provides integration of WebTest with aiohttp.web applications";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:25]:
```suggestion
    description = "Provides integration of WebTest with aiohttp.web applications";
```

File: pkgs/development/python-modules/marshmallow/default.nix [hunk 13]
```diff
@@ -21,7 +19,5 @@ buildPythonPackage rec {
     sha256 = "35ee2fb188f0bd9fc1cf9ac35e45fd394bd1c153cee430745a465ea435514bd5";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/marshmallow/default.nix:19]: Would you mind adding yourself as a maintainer?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/webargs/default.nix (hunk 7 above)
Comment (cript0nauta) [reply pkgs/development/python-modules/webargs/default.nix:37]: Done. BTW, I used nixfmt to format this file, that's why it had an extra linebreak.

File: pkgs/development/python-modules/webargs/default.nix (hunk 9 above)
Comment (cript0nauta) [reply pkgs/development/python-modules/webargs/default.nix:18]: This file was already removed in webargs 8.0.0, so there is no need to remove it from the Nix expressoin.

File: pkgs/development/python-modules/webtest-aiohttp/default.nix (hunk 11 above)
Comment (cript0nauta) [reply pkgs/development/python-modules/webtest-aiohttp/default.nix:17]: Done

File: pkgs/development/python-modules/marshmallow/default.nix (hunk 13 above)
Comment (cript0nauta) [reply pkgs/development/python-modules/marshmallow/default.nix:19]: Done

**Comment by cript0nauta** [comment]
Rebased, upgraded to webargs 8 and fixed suggestions.

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sloria";
+    repo = pname;
+    rev = version;
+    sha256 = "1apr1x0wmnc6l8wv67z4dp00fiiygda6rwpxlspfk7nk9zz37q2j";
+  };
+
+  propagatedBuildInputs = [ webtest ];
+  checkInputs = [ invoke flake8 aiohttp pytest pytest-aiohttp pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:17]:
```suggestion
  checkInputs = [ invoke flake8 aiohttp pytest-aiohttp pytestCheckHook ];
```

File: pkgs/development/python-modules/webtest-aiohttp/default.nix [hunk 15]
```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage, fetchFromGitHub, lib, isPy27, webtest, invoke, flake8
+, aiohttp, pytest, pytest-aiohttp, pytestCheckHook }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/webtest-aiohttp/default.nix:2]:
```suggestion
, aiohttp, pytest-aiohttp, pytestCheckHook }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please add pythonImportsChecks.

**Comment by cript0nauta** [comment]
> Please add pythonImportsChecks.

Done

**Comment by SuperSandro2000** [comment]
I forgot to mention that the last time but please squash the commits into one per package.

**Comment by cript0nauta** [comment]
Sure, done!

**Pull request merged**